### PR TITLE
Update the nav elements active status using Pathname

### LINKED ISSUE #17: Navbar not updating the active state on click
### Description

In the navbar when navigating to different routes , the active state of the element is not changed . Currently in code `aria-current` is used . If using that only, we want to update the state, we have to force render the component. Instead we can use  `useState` and update the active state correctly.  The problem is shown in the screenshot below.

### Screenshots

Screenshot 2024-07-12 225009

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Muhammad-Owais-Warsi**
## Changes proposed
 - Used `usePathname` to get the current path.
 - Using that current path , I update the tailwind dynamically.

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2024-07-13 004356

### CODE REVIEW

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 

Why is there `"use client"` everywhere?

**Comment by Muhammad-Owais-Warsi** [comment]
> Thank you for the contribution 👍
> 
> Why is there `"use client"` everywhere?

Ah, actually i was testing some other functions too. I'll correct them... Thanks for letting me know

**Comment by Muhammad-Owais-Warsi** [comment]
@eddiejaoude on removing the `use client` from those places raise errors. Like `Object.groupBy` is not a function etc..

**Comment by eddiejaoude** [comment]
Please go through and check your changes, "use client" should not be required. I would recommend checking your local node version as this might be causing the problem.

**Comment by Muhammad-Owais-Warsi** [comment]
@eddiejaoude please review and let me know if any changes required.

**Comment by eddiejaoude** [comment]
@Muhammad-Owais-Warsi please recheck each of your changes and make sure you are happy with them, because there are so many unnecessary changes

**Comment by Muhammad-Owais-Warsi** [comment]
@eddiejaoude to me it seems to be fine.

**Comment by eddiejaoude** [comment]
I will comment a few examples in line

File: src/app/repo/status/page.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import List from "@/components/List";
 import checks from "@/utils/checks/index";
 import Stats from "@/components/Stats";
 
+
```
Comment (eddiejaoude) [inline src/app/repo/status/page.js:15]: I don't think this change is required?

File: src/app/repo/status/page.js [hunk 2]
```diff
@@ -195,7 +196,7 @@ const repo = {
   subscribers_count: 54,
 };
 
-const results = checks(repo);
+  const results = checks(repo);
```
Comment (eddiejaoude) [inline src/app/repo/status/page.js:199]: I don't think this change is required?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for making further changes, but more are still required. I have left some example inline comments

Please recheck that all changes in your PR are changes you are happy with, such as extra line changes resulting in files being included in your PR that should not be

**Comment by Muhammad-Owais-Warsi** [comment]
> Thank you for making further changes, but more are still required. I have left some example inline comments
> 
> Please recheck that all changes in your PR are changes you are happy with, such as extra line changes resulting in files being included in your PR that should not be

Sorry for the inconvenience, I'll recheck and let you know.

**Comment by Muhammad-Owais-Warsi** [comment]
@eddiejaoude Please review , I don't think anything should cause any problem now

**Comment by eddiejaoude** [comment]
Thank you for your contribution. As there were still unnecessary changes this PR could not be merge, and now as there are conflicts also, I have made a PR to solve this issue https://github.com/EddieHubCommunity/HealthCheck/pull/43, so I will close this PR.

Hopefully my PR will highlight the changes required, for example the existing condition was the main change only required
https://github.com/EddieHubCommunity/HealthCheck/pull/43/files#diff-6a1559e4b25e80b6a3b7c9c6b7d24db73177617ee8efbc935fae6fd8a59d1667R170

**Comment by Muhammad-Owais-Warsi** [comment]
Thanks @eddiejaoude , will try to improve next time

**Comment by eddiejaoude** [comment]
Thanks @Muhammad-Owais-Warsi for the collaboration

**Pull request closed without merging**